app/models/user_subscription.rb

```diff
@@ -4,12 +4,30 @@
 class UserSubscription < ApplicationRecord
   ALLOWED_TYPES = %w[Article].freeze
 
-  belongs_to :author, class_name: "User", inverse_of: :user_subscriptions
-  belongs_to :subscriber, class_name: "User", inverse_of: :user_subscriptions
+  belongs_to :author, class_name: "User", inverse_of: :source_authored_user_subscriptions
+  belongs_to :subscriber, class_name: "User", inverse_of: :subscribed_to_user_subscriptions
   belongs_to :user_subscription_sourceable, polymorphic: true
 
   validates :author_id, presence: true
-  validates :subscriber_id, presence: true, uniqueness: { scope: %i[user_subscription_sourceable_type user_subscription_sourceable_id] }
+  validates :subscriber_email, presence: true
+  validates :subscriber_id, presence: true, uniqueness: { scope: %i[subscriber_email user_subscription_sourceable_type user_subscription_sourceable_id] }
```

Comment:
Shall we just pull out the this array into a constant and freeze it (similar to `ALLOWED_TYPES`)?

Reply:
This is a little weird...I tried it with:
```ruby
SUBSCRIBER_ID_UNIQUE_SCOPE = %i[
  subscriber_email
  user_subscription_sourceable_type
  user_subscription_sourceable_id
].freeze

validates :subscriber_id, presence: true, uniqueness: { scope: SUBSCRIBER_ID_UNIQUE_SCOPE }
```

And Rubocop complains with:
```
Rails/UniqueValidationWithoutIndex: Uniqueness validation should be with a unique index.
```

I ensured my spelling, changed indentations, etc.

🤔   the index is there, but it only complains when I use a constant or variable. I didn't find anything as to _why_ this is. And I don't want to disable the rule around this line because it's a performance check. I'm going to leave it as is and I've made a note to circle back some time to learn more about why.

Reply:
Rubocop uses static analysis for that cop, so the names need to be inlined, thus I would leave it as it is